### PR TITLE
Redesign of extension list

### PR DESCRIPTION
**Pull request by pascalberger**
Redesign of extension page based on mockup in #1096. 

* Adds a page listing all addins
* Client side search functionality on name and categories
* Removes separate pages for addin categories

image

This is not full implementation of #1096 yet or any additional functionality. Remaining parts will follow in separate PRs.

### CODE REVIEW

File: input/_ExtensionsList.cshtml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+@model IEnumerable<IDocument>
+
+<div class="alert alert-info">
+    <p>
+        This page contains extensions for Cake.
+        If you want to file an issue for any extension listed here, contact the author of the addin.
```
Comment (gep13) [inline input/_ExtensionsList.cshtml:6]: You mention addin wrote the end of this line, is that intended?

File: input/_ExtensionsList.cshtml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    </p>
+    <p>
+        If you're an addin author and want your addin listed here, follow the instructions <a href="https://github.com/cake-build/website/blob/master/README.md#addins" target="_blank">here</a>.
```
Comment (gep13) [inline input/_ExtensionsList.cshtml:9]: Also here...

File: input/_ExtensionsList.cshtml (hunk 1 above)
Comment (pascalberger) [reply input/_ExtensionsList.cshtml]: Nope, chaned to "extension" (at the moment it still only shows addins though)

**Review by gep13 — APPROVED** [review]
LGTM

**Comment by gep13** [comment]
@pascalberger looks like a good start!

**Pull request merged**